pkg/procinfo/procinfo.go

```diff
@@ -59,15 +64,34 @@ func ParseProcessContext(ctx []byte, containers *containers.Containers) (Process
 	procCtx.StartTime = int(binary.LittleEndian.Uint64(ctx[0:8]))
 	cgroupId := binary.LittleEndian.Uint64(ctx[8:16])
 	procCtx.ContainerID = containers.GetCgroupInfo(cgroupId).ContainerId
-	procCtx.Pid = binary.LittleEndian.Uint32(ctx[16:20])
-	procCtx.Tid = binary.LittleEndian.Uint32(ctx[20:24])
-	procCtx.Ppid = binary.LittleEndian.Uint32(ctx[24:28])
-	procCtx.HostTid = binary.LittleEndian.Uint32(ctx[28:32])
-	procCtx.HostPid = binary.LittleEndian.Uint32(ctx[32:36])
-	procCtx.HostPpid = binary.LittleEndian.Uint32(ctx[36:40])
-	procCtx.Uid = binary.LittleEndian.Uint32(ctx[40:44])
-	procCtx.MntId = binary.LittleEndian.Uint32(ctx[44:48])
-	procCtx.PidId = binary.LittleEndian.Uint32(ctx[48:52])
+	decoder := bufferdecoder.New(ctx[16:]) // this is the offset after the cgroup and startTime in the ctx byte array
+	if err := decoder.DecodeUint32(&procCtx.Pid); err != nil {
+		return ProcessCtx{}, err
+	}
+	if err := decoder.DecodeUint32(&procCtx.Tid); err != nil {
+		return ProcessCtx{}, err
+	}
+	if err := decoder.DecodeUint32(&procCtx.Ppid); err != nil {
+		return ProcessCtx{}, err
+	}
+	if err := decoder.DecodeUint32(&procCtx.HostTid); err != nil {
+		return ProcessCtx{}, err
+	}
+	if err := decoder.DecodeUint32(&procCtx.HostPid); err != nil {
+		return ProcessCtx{}, err
+	}
+	if err := decoder.DecodeUint32(&procCtx.HostPpid); err != nil {
+		return ProcessCtx{}, err
+	}
+	if err := decoder.DecodeUint32(&procCtx.Uid); err != nil {
+		return ProcessCtx{}, err
+	}
+	if err := decoder.DecodeUint32(&procCtx.MntId); err != nil {
+		return ProcessCtx{}, err
+	}
+	if err := decoder.DecodeUint32(&procCtx.PidId); err != nil {
+		return ProcessCtx{}, err
+	}
```

Comment:
What's the advantage in using the decoder here?
If you defined the struct in the decoder protocol, that would have been another case, but now the code is longer, and no real advantage here.